api/resource/resource.go

```diff
@@ -101,7 +101,6 @@ func (r *Resource) CopyMergeMetaDataFieldsFrom(other *Resource) error {
 }
 
 func (r *Resource) copyKustomizeSpecificFields(other *Resource) {
-	r.options = other.options
```

Comment:
I think this is subtly different than the previous behavior. Before, when performing a merge, the “other” object’s options completely override those of the receiver. But annotations are merged. So if the receiver has options and the “other” doesn’t, before that would presumably clear the options and after the options would be retained. That might be what's happening in our mysterious ConfigMap test case?

Reply:
Added a line - it's now `r.SetOptions(other.getOptions`) which seemed to fix the ConfigMap case.

Reply:
Great! Should we move it up to `CopyMergeMetaDataFieldsFrom` field though? It's not a Kustomize-specific field anymore.

Looking at that code, I also wonder if we could avoid resetting the annotations repeatedly by making a version of `mergeStringMaps` that is build-annotation-aware.

Reply:
I moved it to `CopyMergeDataFieldsFrom`. 

> Looking at that code, I also wonder if we could avoid resetting the annotations repeatedly by making a version of mergeStringMaps that is build-annotation-aware

That's a good idea, I added one called `mergeStringMapsWithBuildAnnotations`, lmk what you think.